reflex/components/plotly/plotly.py

```diff
@@ -11,6 +13,76 @@
     Figure = Any  # type: ignore
 
 
+def _event_data_signature(e0: Var) -> List[Any]:
+    """For plotly events with event data and no points.
+
+    Args:
+        e0: The event data.
+
+    Returns:
+        The event key extracted from the event data (if defined).
+    """
+    return [Var.create_safe(f"{e0}?.event")]
+
+
+def _event_points_data_signature(e0: Var) -> List[Any]:
+    """For plotly events with event data containing a point array.
+
+    Args:
+        e0: The event data.
+
+    Returns:
+        The event data and the extracted points.
+    """
+    return [
+        Var.create_safe(f"{e0}?.event"),
+        Var.create_safe(
+            f"extractPoints({e0}?.points)",
+        ),
+    ]
+
+
+class _ButtonClickData(Base):
+    """Event data structure for plotly UI buttons."""
+
+    menu: Any
+    button: Any
+    active: Any
+
+
+def _button_click_signature(e0: _ButtonClickData) -> List[Any]:
+    """For plotly button click events.
+
+    Args:
+        e0: The button click data.
+
+    Returns:
+        The menu, button, and active state.
+    """
+    return [e0.menu, e0.button, e0.active]
+
+
+def _passthrough_signature(e0: Var) -> List[Any]:
```

Comment:
Wouldn't the `_passthrough_signature` be useful for more than just plotly?

I'm wondering if we should have a `signature.py` for all "signature presets" (maybe we can do that in later PR tho)